extensions/ftp/src/main/java/org/mule/extension/ftp/internal/sftp/connection/SftpClient.java

```diff
@@ -211,14 +212,15 @@ private void configureSession(String user) throws JSchException
 
     private void configureHostChecking(Properties hash) throws JSchException
     {
-        if (getKnownHostsFile() == null)
+        if (knownHostsFile == null)
         {
             hash.put(STRICT_HOST_KEY_CHECKING, "no");
         }
         else
         {
+            checkExists(knownHostsFile);
             hash.put(STRICT_HOST_KEY_CHECKING, "ask");
-            jsch.setKnownHosts(getKnownHostsFile().getAbsolutePath());
+            jsch.setKnownHosts(knownHostsFile);
```

Comment:
before it was the absolute path of a File object. Now I shift it so that the client only has the string of such path and I control absoluteness and existence on the provider.

It's the same behaviour, just organized differently